It looks like the problem is in stopping of the servo. I have a stop method in the servo class, but probably the main OPP platform is shutting down before the OPPServo.stop() can complete. How do I tell the main platform to wait?

If I remove the stop method for the servo, the tests all pass. But I want a stop method to command the servo to disable before MPF shuts down. This did work on actual hardware and I am not sure what the error means. Here is the error:

`sudo python3 -m unittest test_OPP
..Failed to stop all tasks
E
======================================================================
ERROR: testOpp (test_OPP.TestOPPStm32)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/mpf/tests/test_OPP.py", line 166, in tearDown
    super().tearDown()
  File "/usr/local/lib/python3.6/dist-packages/mpf/tests/MpfTestCase.py", line 958, in tearDown
    self.machine._do_stop()
  File "/usr/local/lib/python3.6/dist-packages/mpf/core/machine.py", line 729, in _do_stop
    self.shutdown()
  File "/usr/local/lib/python3.6/dist-packages/mpf/core/machine.py", line 774, in shutdown
    self.clock.loop.close()
  File "/usr/local/lib/python3.6/dist-packages/mpf/tests/loop.py", line 286, in close
    open_tasks = [t for t in tasks if (not t.done() and not isinstance(t.get_coro(), asyncio.Lock))]
  File "/usr/local/lib/python3.6/dist-packages/mpf/tests/loop.py", line 286, in <listcomp>
    open_tasks = [t for t in tasks if (not t.done() and not isinstance(t.get_coro(), asyncio.Lock))]
AttributeError: '_asyncio.Task' object has no attribute 'get_coro'

----------------------------------------------------------------------
Ran 3 tests in 0.168s

FAILED (errors=1)
`

--- a/mpf/tests/test_OPP.py
+++ b/mpf/tests/test_OPP.py
@@ -161,6 +161,9 @@ class TestOPPStm32(MpfTestCase):
         self.clock.mock_serial("com2", self.serialMocks["com2"])
 
     def tearDown(self):
+        self.serialMocks["com1"].expected_commands[
+            self._crc_message(b'\x20\x40\x30\x00\x00\x01\x00\x00\x00', False)] = False
+
         for port, mock in self.serialMocks.items():
             self.assertFalse(mock.crashed, "Mock {} crashed".format(port))
         super().tearDown()

